src/test/sql/sql_pipeline_statement_test.cpp

```diff
@@ -683,4 +692,27 @@ TEST_F(SQLPipelineStatementTest, CacheQueryPlan) {
   EXPECT_TRUE(cache.has(_select_query_a));
 }
 
+TEST_F(SQLPipelineStatementTest, RecreateSubselectFromCache) {
+  const std::string subselect_query = "SELECT * FROM table_int WHERE a = (SELECT MAX(b) FROM table_int)";
+
+  SQLPipelineStatement first_subselect_sql_pipeline{subselect_query};
+  const auto first_subselect_result = first_subselect_sql_pipeline.get_result_table();
+
+  auto expected_first_result = std::make_shared<Table>(_int_int_int_column_definitions, TableType::Data);
+  expected_first_result->append({10, 10, 10});
+
+  EXPECT_TABLE_EQ_UNORDERED(first_subselect_result, expected_first_result);
+
+  SQLPipelineStatement{"INSERT INTO table_int VALUES (11, 11, 11)"}.get_result_table();
+
+  SQLPipelineStatement second_subselect_sql_pipeline{subselect_query};
+  const auto second_subselect_result = second_subselect_sql_pipeline.get_result_table();
```

Comment:
This test is a bit too high-level. You are assuming that the sql pipeline has a cache.
The bug was with the recreate function, so you should explicitly use that one instead of relying on the cache to call it for you.
Other than that, this is good to go.

Reply:
There is recreation_test.cpp (gosh, what a name), test should probably go there.